File: C 26.08.2022/SampleTests.c
/*
  The sample test suite is developed by the following CodeWars user: https://www.codewars.com/users/porkfreezer
*/

#include <criterion/criterion.h>

char *find_the_number_plate(int id);

static void do_test(int id, const char *plate) {
  char *actual = find_the_number_plate(id);
  cr_assert_str_eq(plate, actual, "Expected %s for %d, but got %s", plate, id, actual);
  if (actual)
    free(actual);
}

Test(number_plates, should_pass_basic_tests) {
  do_test(3, "aaa004");
  do_test(1487, "baa489");
  do_test(40000, "oba041");
  do_test(17558423, "zzz999");
  do_test(234567, "aja802");
  do_test(43056, "rba100");
}

File: Rust 19.08.2022/SampleTests.rs
/*
  The sample test suite is developed by the following CodeWars user: www.codewars.com/users/akar-0
*/


// See https://doc.rust-lang.org/stable/rust-by-example/testing/unit_testing.html

#[cfg(test)]
mod tests {
    use super::find_the_number_plate;
        
    fn dotest(n: u32, expected: &str) {
        let actual = find_the_number_plate(n);
        assert!(actual == expected, 
            "With n = {n}\nExpected \"{expected}\" but got \"{actual}\"")
    }

    #[test]
    fn fixed_tests() {
        dotest(3,"aaa004");
        dotest(1487,"baa489");
        dotest(40000,"oba041");
        dotest(17558423,"zzz999");
        dotest(43056,"rba100");
        dotest(234567,"aja802");
    }
}
